st2api/tests/unit/controllers/v1/test_executions_auth.py

```diff
@@ -145,3 +198,30 @@ def test_post_with_st2_context_in_headers(self):
         self.assertEqual(resp.status_int, 201)
         self.assertEqual(resp.json['context']['user'], 'tokenuser')
         self.assertEqual(resp.json['context']['parent'], context['parent'])
+
+    @mock.patch.object(ApiKey, 'get', mock.Mock(return_value=TEST_APIKEY))
+    @mock.patch.object(User, 'get_by_name', mock.Mock(return_value=TEST_USER))
+    def test_template_encrypted_params_with_apikey(self):
+        resp = self._do_post(LIVE_ACTION_DEFAULT_ENCRYPT, headers={
+            'St2-Api-key': 'secret_key'
+        })
+        self.assertEqual(resp.status_int, 201)
+        self.assertEqual(resp.json['parameters']['encrypted_param'], 'foo')
```

Comment:
I guess we should probably document this - "behavior". Not related to your change directly, but those tests exposed this "potential issue" which could bite users which are not aware of this behavior.

Something along the lines of "If encrypted system datastore parameter is used for action parameter default value the unecrypted version of the value will be visible to any user which has permission to execute that action. If the raw value shouldn't be exposed you should use alternative means like retrieve encrypted value from the datastore inside the workflow."

An "alternative" would also be to modify the behavior on our side - don't expose raw value to users. Only allow admins to see them when using show secrets.

EDIT: After some more thought - this should probably already work as expected as long as the parameter is also marked as ``secret: true``. Can you please add another test case for that? Same as the existing ones, but with the parameters additionally being marked as ``secret: true``.

Reply:
I've added a test case for it here - 58326dfdb.

I will also push a corresponding change. As noted in the commit message and here - using ``decrypt_kv`` on parameter which is not marked as secret could be dangerous and rarely makes sense.

In the future, we could also add a "lint" rule for that to ``st2-register-content`` which would warn user if such parameter is found.